.github/workflows/test.yaml

```diff
@@ -1,24 +1,67 @@
-name: Run XYZ Hub tests
+name: Run Naksha tests
 
-on: [push, pull_request, workflow_dispatch]
+on:
+  pull_request:
+    types:
+      - opened       # run whenever PR is opened
+      - synchronize  # run whenever PR is updated
+      - reopened     # run whenever PR is reopoened
+  workflow_dispatch: # manual run
+
+
+permissions:
+  checks: write         # for junit reporting
+  pull-requests: write  # for jacoco PR comments
+
+env:
+  MIN_COVERAGE_OVERALL: 25
+  MIN_COVERAGE_CHANGED_FILES: 80
 
 jobs:
   test:
     runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgis/postgis # Postgres with PostGIS extension
+        env:
+          POSTGRES_PASSWORD: password
+          POSTGRES_USER: postgres
+          POSTGRES_DB: postgres
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
     steps:
-      - uses: actions/checkout@v2
-      - name: Cache local Maven repository
-        uses: actions/cache@v2
-        with:
-          path: ~/.m2/repository
-          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
-          restore-keys: |
-            ${{ runner.os }}-maven-
+      - uses: actions/checkout@v4
       - name: Setup Java
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: '17'
+      - name: Setup Gradle, execute tests & verify overall coverage
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: test jacocoTestReport jacocoTestCoverageVerification
```

Comment:
Can we simply use "jacocoTestCoverageVerification" , if it can internally execute other goals automatically?